src/test/run-pass/issue-29037.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::rc::Rc;
```

Comment:
please add a comment explaining what this is testing.

(I know it is testing that for each of the `P<X>` types, `P` is covariant in `X`, just write a comment saying as much, so that someone doesn't have to go looking up the issue.)